File: soil_health.py
def analyze_soil_health(nitrogen, phosphorus, potassium, ph):

    nutrients = {
        "Nitrogen": nitrogen,
        "Phosphorus": phosphorus,
        "Potassium": potassium,
        "pH": ph
    }

    score = 100
    issues = []

    if nitrogen < 40:
        score -= 15
        issues.append("Nitrogen is low.")
    elif nitrogen > 80:
        score -= 10
        issues.append("Nitrogen is high.")

    if phosphorus < 40:
        score -= 15
        issues.append("Phosphorus is low.")
    elif phosphorus > 80:
        score -= 10
        issues.append("Phosphorus is high.")

    if potassium < 40:
        score -= 15
        issues.append("Potassium is low.")
    elif potassium > 80:
        score -= 10
        issues.append("Potassium is high.")

    if ph < 5.5:
        score -= 15
        issues.append("Soil is acidic.")
    elif ph > 8.0:
        score -= 15
        issues.append("Soil is alkaline.")

    if score >= 80:
        health = "Excellent"
    elif score >= 60:
        health = "Good"
    elif score >= 40:
        health = "Moderate"
    else:
        health = "Poor"

    return {
        "score": score,
        "health": health,
        "nutrients": nutrients,
        "issues": issues
    }

File: soil_health.js
export function analyzeSoilHealth(nitrogen, phosphorus, potassium, ph) {
  const nutrients = {
    Nitrogen: nitrogen,
    Phosphorus: phosphorus,
    Potassium: potassium,
    pH: ph
  };

  let score = 100;
  const issues = [];

  if (nitrogen < 40) {
    score -= 15;
    issues.push("Nitrogen is low.");
  } else if (nitrogen > 80) {
    score -= 10;
    issues.push("Nitrogen is high.");
  }

  if (phosphorus < 40) {
    score -= 15;
    issues.push("Phosphorus is low.");
  } else if (phosphorus > 80) {
    score -= 10;
    issues.push("Phosphorus is high.");
  }

  if (potassium < 40) {
    score -= 15;
    issues.push("Potassium is low.");
  } else if (potassium > 80) {
    score -= 10;
    issues.push("Potassium is high.");
  }

  if (ph < 5.5) {
    score -= 15;
    issues.push("Soil is acidic.");
  } else if (ph > 8.0) {
    score -= 15;
    issues.push("Soil is alkaline.");
  }

  let health = "Poor";
  if (score >= 80) {
    health = "Excellent";
  } else if (score >= 60) {
    health = "Good";
  } else if (score >= 40) {
    health = "Moderate";
  }

  return {
    score,
    health,
    nutrients,
    issues
  };
}
